### DISCUSSION TITLE
How can I open a file?

### DISCUSSION THREAD

**Discussion by webysther** [coder/code-server#6224] [Q&A]
I tried this solution [here](https://superuser.com/questions/1709604/how-can-i-open-a-file-with-a-chrome-web-app) to select a extension and open direct on coder server but dont works, i know the API on chrome is stable now but I can't find a solution, any directions?

**Comment by webysther** [comment]
Looks like the command don't work:

```sh
"C:\Program Files\Google\Chrome\Application\chrome_proxy.exe" 
--profile-directory=Default 
--app-id=jhajnhdfkmhndfgenjbbhfohenlappdi 
"PATH/FILE.EXTENSION"
```

If have some strategy to make this work maybe create a "open with..." become a reality:

```ini
Windows Registry Editor Version 5.00

; Open files [HKEY_CLASSES_ROOT\*\shell\Open with VS Code] @="Edit with VS Code"
"Icon"="C:\\Program Files\\Microsoft VS Code\\Code.exe,0"

[HKEY_CLASSES_ROOT\*\shell\Open with VS Code\command]
@="\"C:\\Program Files\\Microsoft VS Code\\Code.exe\" \"%1\""

; This will make it appear when you right click ON a folder
; The "Icon" line can be removed if you don't want the icon to appear
[HKEY_CLASSES_ROOT\Directory\shell\vscode]
@="Open in VS Code"
"Icon"="\"C:\\Program Files\\Microsoft VS Code\\Code.exe\",0"

[HKEY_CLASSES_ROOT\Directory\shell\vscode\command]
@="\"C:\\Program Files\\Microsoft VS Code\\Code.exe\" \"%1\""

; This will make it appear when you right click INSIDE a folder
; The "Icon" line can be removed if you don't want the icon to appear
[HKEY_CLASSES_ROOT\Directory\Background\shell\vscode]
@="Open in VS Code"
"Icon"="\"C:\\Program Files\\Microsoft VS Code\\Code.exe\",0"

[HKEY_CLASSES_ROOT\Directory\Background\shell\vscode\command]
@="\"C:\\Program Files\\Microsoft VS Code\\Code.exe\" \"%V\""
```
[source](https://www.codeproject.com/Tips/5296005/How-to-Open-Folder-Directly-in-Visual-Studio-Code)

**Comment by code-asher** [comment]
It looks like this is for native VS Code in which case the repo you want is this one: https://github.com/microsoft/vscode/

**Reply by webysther** [reply]
I want use the code-server as my local editor.

**Reply by code-asher** [reply]
Ah I see, I was confused by the paths to native VS Code in the comment.

code-server technically has no Windows support right now but if you are able to get it running then open code-server in your browser and you can use `code-server -r /path/to/file` and it will open in the current instance displaying in your browser (this should work both inside the integrated terminal and in external terminals).

There is also an `--open` flag which will automatically launch the browser but I think this only works when first launching code-server.  We should probably support it when opening files in an already-running instance of code-server.

Let me know if I have misunderstood what you are trying to do though.

**Reply by code-asher** [reply]
I opened an issue for `--open` here https://github.com/coder/code-server/issues/6252

**Reply by webysther** [reply]
> Ah I see, I was confused by the paths to native VS Code in the comment.
> 
> code-server technically has no Windows support right now but if you are able to get it running then open code-server in your browser and you can use `code-server -r /path/to/file` and it will open in the current instance displaying in your browser (this should work both inside the integrated terminal and in external terminals).
> 
> There is also an `--open` flag which will automatically launch the browser but I think this only works when first launching code-server. We should probably support it when opening files in an already-running instance of code-server.
> 
> Let me know if I have misunderstood what you are trying to do though.

Half way, I need to be clear, english isn't my first language, sorry.

I have a code-server running in the homelab, works so well and greatly that I decide remove my local vs code, everything works, but I not able to open files with dual click ou using a option like "open with...", my main system is windows... At the first explanation I discovered that google chrome apps works a litle bit more integrated to windows.

Is possible to open a local file just drop in the code server and works really well, but have integrated to OS (linux or windows) will the nice from a version running in a browser in a remote server.

To this works is straightforward you need to support receive this request from the browser by the code server:

```shell
"C:\Program Files\Google\Chrome\Application\chrome_proxy.exe" 
--profile-directory=Default 
--app-id=jhajnhdfkmhndfgenjbbhfohenlappdi 
"PATH/FILE.EXTENSION"
```

The app-id is the code-server [running as PWA](https://medium.com/progressivewebapps/chrome-now-supports-pwa-install-experience-551cf896c0f9) and when you put the `"PATH/FILE.EXTENSION"` this is sent to PWA and works in few webapps but not in code server.

If this works I was able to registry the PWA as a open with and default open app for some specific extensions of files.

**Reply by webysther** [reply]
Few application registry a custom open and in case of code server maybe "code-server://" and this was make more wide integration.

**Reply by code-asher** [reply]
Ahh I see!  That would be cool.  Adding a protocol handler for `code-server://` seems like a nice way to implement it.  Related: https://github.com/coder/code-server/issues/1571
